src/targets/resources/rock.js

```diff
@@ -9,10 +13,46 @@ export class Rock extends Resource {
     constructor(data) {
         data.skill = "mining";
         super(data);
+        this.requiredMiningLevel = data.requiredMiningLevel;
     }
 
     isClickable() {
         let curWeapon = this.scene.dashboard.equipment.equipment.WEAPON;
-        return curWeapon.item == "Pickaxe";
+        let inventory = this.scene.dashboard.inventory;
+        let chat = this.scene.scene.get(CONSTANTS.SCENES.CHAT);
+
+        let i = inventory.getKeywordInInventory("Pickaxe");
+        let miningLevel = calcLevel(characterData.getSkillXp("mining"));
+        let canUsePickaxe = false;
+
+        if (curWeapon.item == "Pickaxe") {
+            canUsePickaxe = miningLevel >= curWeapon.requiredMiningLevel;
+        } else if (i != -1) {
+            canUsePickaxe =
+                miningLevel >=
+                getItemClass(characterData.getInventory()[i].item, this.scene)
+                    .requiredMiningLevel;
+        } else {
+            chat.writeText("This action requires a pickaxe.");
+            return false;
+        }
+
+        // Currently a small bug here that will not let you mine if the pickaxe you
+        // are unable to use is higher in the inventory than another pickaxe that you can use
```

Comment:
Would it be hard to loop through and find all useable pickaxes and then take the one with the highest level?

Reply:
updated! https://github.com/EthanVieira/rs-clicker/pull/57/commits/8048b8819e2c3b3ec52cc513a0ede33b9f57569a